SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimAnalyzer.cc

```diff
@@ -261,7 +460,522 @@ EcalEBTrigPrimAnalyzer::analyze(const edm::Event& iEvent, const  edm::EventSetup
 	hAllRechitEt_ ->Fill(rhEt);
     }
   }
+
+
+
+  //  float Etsum=0;
+  std::vector<SimpleCluster> cluCollection;
+  std::vector<SimpleCaloHit2> myHits;
+  if ( recHits_ ) {    
+
+    myHits.clear();
+    for (unsigned int j=0;j<rechit_EB_col.product()->size();j++) {
+      const EBDetId & rhID=(*rechit_EB_col.product())[j].detid();
+      float eta =  theBarrelGeometry->getGeometry(rhID)->getPosition().eta();
+      float energy= (*rechit_EB_col.product())[j].energy();
+      float et= energy/cosh(eta);
+      if (energy <=0) continue;
+      if (debug_) std::cout << " my Hits energy " << energy << " et " << et << std::endl;
+      SimpleCaloHit2 hit(et);
+      hit.setPosition(GlobalVector( theBarrelGeometry->getGeometry(rhID)->getPosition().x(), 
+				    theBarrelGeometry->getGeometry(rhID)->getPosition().y(), 
+				    theBarrelGeometry->getGeometry(rhID)->getPosition().z()));
+
+
+
+      hit.setId(rhID);
+      myHits.push_back(hit);	  
+
+
+    }
+    if ( debug_) std::cout << " myHits collection size " << myHits.size() << std::endl;
+    for (unsigned int iRH=0;iRH<myHits.size();++iRH) {
+      if (debug_) std::cout << " RecHit Et " << myHits[iRH].et() << std::endl;
+    }    
+
+
+    /*
+      for  (int ii=0;ii<nXtals_;ii++) {
+      const EBDetId & xtalID =  d.crystalsInCluster()[ii];
+      
+      for (unsigned int j=0;j<rechit_EB_col.product()->size();j++) {
+      // check that the recHit correspond to the crystal in the cluster TP
+      const EBDetId & rhID=(*rechit_EB_col.product())[j].id();
+      float theta =  theBarrelGeometry->getGeometry(rhID)->getPosition().theta();
+      if ( rhID == xtalID ) {
+      // std::cout << " RH id " << rhID << std::endl;
+      Etsum+=((*rechit_EB_col.product())[j].energy())*sin(theta);
+      }
+      }
+      }
+    */
+    
+    // if rechits are available  
+    cluCollection = makeCluster ( myHits, 3, 3  );
+    
+  }  // if (recHIts_)
+  
+  if (debug_) std::cout << " cluCollection from RH " << cluCollection.size()  << std::endl;  
+  
+  
+  // loop over cluster TP
+  int nClu=0;
+  int nClu_Etgt_05=0;
+  int nClu_Etgt_1=0;
+  int nClu_Etgt_2=0;
+  int nClu_Etgt_3=0;
+
+
+
+  if (tpCluHandleIsValid ) {
+ 
+    nCl_ = tpClu.product()->size();
+    int nCluTPPerEvt=0;
+    for (unsigned int i=0;i<tpClu.product()->size();i++) {
+      EcalEBClusterTriggerPrimitiveDigi d=(*(tpClu.product()))[i];
+      const EBDetId TPid= d.id();
+      float etaCluTP =  theBarrelGeometry->getGeometry(TPid)->getPosition().eta();
+      float phiCluTP =  theBarrelGeometry->getGeometry(TPid)->getPosition().phi();
+      //std::cout << " cluTP eta " << d.eta() << " " << d.phi() << std::endl;
+      nXtals_= d.crystalsInCluster().size();
+      nCluTPPerEvt++;
+      nTotCluTP_++;
+
+      /*
```

Comment:
Delete commented out code, or justify with a comment why you still need to keep it